spot_rl_experiments/spot_rl/utils/helper_nodes.py

```diff
@@ -24,28 +25,36 @@ def __init__(self, spot):
         rospy.init_node("spot_ros_proprioception_node", disable_signals=True)
         self.spot = spot
 
-        # Instantiate filtered image publishers
+        # Publishers
         self.pub = rospy.Publisher(rt.ROBOT_STATE, Float32MultiArray, queue_size=1)
-        self.last_publish = time.time()
-        rospy.loginfo("[spot_ros_proprioception_node]: Publishing has started.")
+        self.odom_broadcaster = rospy.Publisher(
+            rt.ODOM_TOPIC, Odometry, queue_size=1, latch=True
+        )
+        self.marker_pub = rospy.Publisher(
+            "gripper_trajectory_marker_array", MarkerArray, queue_size=1
+        )
 
+        self.last_publish = time.time()
         self.nav_pose_buff = None
         self.buff_idx = 0
 
-        # Instantiate static transform broadcaster for publishing spot to spotWorld transform
+        # Static transform broadcaster
         self.static_tf_broadcaster = StaticTransformBroadcaster()
 
-        # broadcast spot odometry to visualize path in rviz
-        self.odom_broadcaster = rospy.Publisher(
-            rt.ODOM_TOPIC, Odometry, queue_size=1, latch=True
-        )
+        self.gripper_traj = []
 
     def publish_msgs(self):
         st = time.time()
         robot_state = self.spot.get_robot_state()
-
         robot_kinematic_snapshot_tree = robot_state.kinematic_state.transforms_snapshot
 
+        position, rotation = self.spot.get_base_transform_to("link_wr1")
+        ee_pose = np.array([position.x, position.y, position.z])
+
+        self.gripper_traj.append(ee_pose)
```

Comment:
what is the motivation for adding this? is that you want to use ros vis tool to visualize the gripper trajectory for debugging?

Reply:
Yes, this publishes the end effector pose as a marker array, which we can visualize in Rviz for understanding its trajectory.